File: Lib/Telerik/Examples/QuickStart/DataEntry/Validation/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.DataEntry.Validation
{
    public class Person : IEditableObject, INotifyPropertyChanged
    {
        EmployeeData backupEmplData;
        private DateTime startingDate;
        private string firstName;
        private string lastName;
        private int salary;
        private OccupationPositions occupation;
        private string phoneNum;

        public enum OccupationPositions
        {
            Casher,
            Consultant,
            Security,
            Supplies,
            SuppliesManager,
            StaffManager,
            HygieneStaff
        }

        public struct EmployeeData
        {
            internal DateTime startingDate;
            internal string firstName;
            internal string lastName;
            internal int salary;
            internal OccupationPositions occupation;
            internal string phoneNum;
        }

        public Person()
        {
            this.firstName = string.Empty;
            this.lastName = string.Empty;
            this.salary = 1500;
            this.phoneNum = string.Empty;
        }

        public Person(DateTime startingDate, string fName, string lName, OccupationPositions occ, string phoneNum, int sal)
        {
            this.backupEmplData = new EmployeeData();
            this.backupEmplData.startingDate = startingDate;
            this.StartingDate = startingDate;
            this.backupEmplData.salary = sal;
            this.Salary = sal;
            this.backupEmplData.firstName = fName;
            this.FirstName = fName;
            this.backupEmplData.lastName = lName;
            this.LastName = lName;
            this.backupEmplData.occupation = occ;
            this.Occupation = occ;
            this.backupEmplData.phoneNum = phoneNum;
            this.PhoneNumber = phoneNum;
            this.StartingDate = startingDate;
        }

        [DisplayName("First Name")]
        public string FirstName
        {
            get
            {
                return firstName;
            }
            set
            {
                firstName = value;
                NotifyPropertyChanged("FirstName");
            }
        }

        [DisplayName("Last Name")]
        public string LastName
        {
            get
            {
                return lastName;
            }
            set
            {
                lastName = value;
                NotifyPropertyChanged("LastName");
            }
        }

        public OccupationPositions Occupation
        {
            get
            {
                return occupation;
            }
            set
            {
                occupation = value;
                NotifyPropertyChanged("Occupation");
            }
        }

        [DisplayName("Starting Date")]
        public DateTime StartingDate
        {
            get
            {
                return startingDate;
            }
            set
            {
                startingDate = value;
                NotifyPropertyChanged("StartingDate");
            }
        }

        [DisplayName("Phone Number")]
        public string PhoneNumber
        {
            get
            {
                return phoneNum;
            }
            set
            {
                phoneNum = value;
                NotifyPropertyChanged("PhoneNumber");
            }
        }

        [RadRange(1500,1700)]
        public int Salary
        {
            get
            {
                return salary;
            }
            set
            {
                salary = value;
                NotifyPropertyChanged("Salary");
            }
        }



        public void BeginEdit()
        {

        }

        public void CancelEdit()
        {
            Salary = this.backupEmplData.salary;
            StartingDate = this.backupEmplData.startingDate;
            FirstName = this.backupEmplData.firstName;
            LastName = this.backupEmplData.lastName;
            Occupation = this.backupEmplData.occupation;
            PhoneNumber = this.backupEmplData.phoneNum;
        }

        public void EndEdit()
        {
            this.backupEmplData.salary = Salary;
            this.backupEmplData.startingDate = StartingDate;
            this.backupEmplData.firstName = FirstName;
            this.backupEmplData.lastName = LastName;
            this.backupEmplData.occupation = Occupation;
            this.backupEmplData.phoneNum = PhoneNumber;
        }

        public event PropertyChangedEventHandler PropertyChanged;

        private void NotifyPropertyChanged(string info)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(info));
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/DataEntry/Validation/Person.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.DataEntry.Validation
    Public Class Person
        Implements IEditableObject
        Implements INotifyPropertyChanged
        Private backupEmplData As EmployeeData
        Private m_startingDate As DateTime
        Private m_firstName As String
        Private m_lastName As String
        Private m_salary As Integer
        Private m_occupation As OccupationPositions
        Private phoneNum As String

        Public Enum OccupationPositions
            Casher
            Consultant
            Security
            Supplies
            SuppliesManager
            StaffManager
            HygieneStaff
        End Enum

        Public Structure EmployeeData
            Friend startingDate As DateTime
            Friend firstName As String
            Friend lastName As String
            Friend salary As Integer
            Friend occupation As OccupationPositions
            Friend phoneNum As String
        End Structure

        '
        Public Sub New()
            Me.m_firstName = String.Empty
            Me.m_lastName = String.Empty
            Me.m_salary = 1500
            Me.phoneNum = String.Empty
        End Sub

        Public Sub New(startingDate As DateTime, fName As String, lName As String, occ As OccupationPositions, phoneNum As String, sal As Integer)
            Me.backupEmplData = New EmployeeData()
            Me.backupEmplData.startingDate = startingDate
            Me.StartingDate = startingDate
            Me.backupEmplData.salary = sal
            Me.Salary = sal
            Me.backupEmplData.firstName = fName
            Me.FirstName = fName
            Me.backupEmplData.lastName = lName
            Me.LastName = lName
            Me.backupEmplData.occupation = occ
            Me.Occupation = occ
            Me.backupEmplData.phoneNum = phoneNum
            Me.PhoneNumber = phoneNum
            Me.StartingDate = startingDate
        End Sub

        <DisplayName("First Name")> _
        Public Property FirstName() As String
            Get
                Return m_firstName
            End Get
            Set(value As String)
                m_firstName = value
                NotifyPropertyChanged("FirstName")
            End Set
        End Property

        <DisplayName("Last Name")> _
        Public Property LastName() As String
            Get
                Return m_lastName
            End Get
            Set(value As String)
                m_lastName = value
                NotifyPropertyChanged("LastName")
            End Set
        End Property

        Public Property Occupation() As OccupationPositions
            Get
                Return m_occupation
            End Get
            Set(value As OccupationPositions)
                m_occupation = value
                NotifyPropertyChanged("Occupation")
            End Set
        End Property

        <DisplayName("Starting Date")> _
        Public Property StartingDate() As DateTime
            Get
                Return m_startingDate
            End Get
            Set(value As DateTime)
                m_startingDate = value
                NotifyPropertyChanged("StartingDate")
            End Set
        End Property

        <DisplayName("Phone Number")> _
        Public Property PhoneNumber() As String
            Get
                Return phoneNum
            End Get
            Set(value As String)
                phoneNum = value
                NotifyPropertyChanged("PhoneNumber")
            End Set
        End Property

        <RadRange(1500, 1700)> _
        Public Property Salary() As Integer
            Get
                Return m_salary
            End Get
            Set(value As Integer)
                m_salary = value
                NotifyPropertyChanged("Salary")
            End Set
        End Property



        Public Sub BeginEdit() Implements IEditableObject.BeginEdit

        End Sub

        Public Sub CancelEdit() Implements IEditableObject.CancelEdit
            Salary = Me.backupEmplData.salary
            StartingDate = Me.backupEmplData.startingDate
            FirstName = Me.backupEmplData.firstName
            LastName = Me.backupEmplData.lastName
            Occupation = Me.backupEmplData.occupation
            PhoneNumber = Me.backupEmplData.phoneNum
        End Sub

        Public Sub EndEdit() Implements IEditableObject.EndEdit
            Me.backupEmplData.salary = Salary
            Me.backupEmplData.startingDate = StartingDate
            Me.backupEmplData.firstName = FirstName
            Me.backupEmplData.lastName = LastName
            Me.backupEmplData.occupation = Occupation
            Me.backupEmplData.phoneNum = PhoneNumber
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Private Sub NotifyPropertyChanged(info As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(info))
        End Sub
    End Class
End Namespace